### PR TITLE
ci: add K8S deployment

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Add K8S deployment:
- add Helm Chart for Server-API with templates:
  - deployment
  - service
  - hpa
  - ingress

@BOLT04 I don't know if you are familiar with k8s, so please don't review if you are not, but if you want to you can 😅

### CODE REVIEW

**Comment by BOLT04** [comment]
@magicmatatjahu I'm a bit familiar with Kubernetes, but I've never used Helm 😅 
I just have one comment 🙂, is there is a missing quote `"` here?

```diff
- | image.repository | string | `asyncapi/server-api"` |  |
+ | image.repository | string | `"asyncapi/server-api"` |  |
```

Other than that this PR seems pretty cool, I only saw the `event-gateway` repo with these Helm definitions too so it would be cool if @smoya could also take a look 👍

File: deployments/k8s/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  echo http://$NODE_IP:$NODE_PORT
+{{- else if contains "LoadBalancer" .Values.service.type }}
+     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+           You can watch the status of by running 'kubectl get --namespace {{ .Release.Namespace }} svc -w {{ include "asyncapi-event-gateway.fullname" . }}'
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ include "asyncapi-event-gateway.fullname" . }} --template "{{"{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"}}")
+  echo http://$SERVICE_IP:{{ .Values.service.port }}
+{{- else if contains "ClusterIP" .Values.service.type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app.kubernetes.io/name={{ include "asyncapi-event-gateway.name" . }},app.kubernetes.io/instance={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
+  export CONTAINER_PORT=$(kubectl get pod --namespace {{ .Release.Namespace }} $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
+  echo "Visit http://127.0.0.1:8080 to use your application"
```
Comment (smoya) [inline deployments/k8s/templates/NOTES.txt:14]: Is this `http://127.0.0.1:8080` a real thing?

File: deployments/k8s/values.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  tag: "latest"
+
+imagePullSecrets: []
+
+replicaCount: 2
+
+# Server API opened ports.
+ports:
+  # Port where the Server API will be available.
+  http: 5000
```
Comment (smoya) [inline deployments/k8s/values.yaml:17]: I think this is going to be `80` finally by default, right? https://github.com/asyncapi/server-api/pull/24/files#diff-841254fe75488c1bd4cd7f68f00b4be0e48dcfbc4a16b45847b68295e0e3b27bR21

File: deployments/k8s/values.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+# Env vars. Needed for configuring the app.
+env: {}
```
Comment (smoya) [inline deployments/k8s/values.yaml:20]: The fact there are no env vars seems like a smell to me. Something we have to work on soon so the whole app config can be set via environment variables. An interesting approach to make saas is https://12factor.net, where their  [config](https://12factor.net) section advocates for environment variables (which I'm totally in as well)

**Comment by smoya** [comment]
There is one thing it is not clear to me within this PR context.

Is this chart supposed to be used by users or rather the one used for deploying api.asyncapi.com (or whatever name we choose) ? Because `values.yaml` might be different on each

**Comment by magicmatatjahu** [comment]
@smoya 

> Is this chart supposed to be used by users or rather the one used for deploying api.asyncapi.com (or whatever name we choose) ? Because values.yaml might be different on each

You should be able to deploy on your machines :) Of course we will use these templates for `api.asyncapi.com` but primary it is for community.

File: deployments/k8s/values.yaml (hunk 3 above)
Comment (magicmatatjahu) [reply deployments/k8s/values.yaml:20]: I know that and I'm all for environment variables too :) Don't worry, we will do it but not in this PR. I can remove the `envs` field but I don't know if it is that important.

File: deployments/k8s/values.yaml (hunk 2 above)
Comment (magicmatatjahu) [reply deployments/k8s/values.yaml:17]: Yes, I forgot to change it :) Thanks!

File: deployments/k8s/values.yaml (hunk 3 above)
Comment (smoya) [reply deployments/k8s/values.yaml:20]: No,it is ok to leave it here.

**Review by smoya — DISMISSED** [review]
LGTM!

**Comment by BOLT04** [comment]
This looks great 👍, I think we can merge this @magicmatatjahu  🚀

**Comment by magicmatatjahu** [comment]
@BOLT04 Thanks but we need to wait for that PR https://github.com/asyncapi/.github/pull/117 and then for Docker Image in another PR in this repo :) I wanna first deploy app with image to check that everything work as expected :) I hope that it will be tomorrow.

File: deployments/k8s/values.yaml (hunk 3 above)
Comment (magicmatatjahu) [reply deployments/k8s/values.yaml:20]: Issue for that is created https://github.com/asyncapi/server-api/issues/39 :)

**Comment by magicmatatjahu** [comment]
@smoya @BOLT04 I improved templates by adding to every resource the namespace. I think that we can merge it. Also, important thing: should we go with `ci` (or `chore`) prefix for PR's title? It's not related with source code but with infra and `infra` prefix isn't allowed and also I don't want release new version of package, or we should, WDYT?

**Comment by smoya** [comment]
> @smoya @BOLT04 I improved templates by adding to every resource the namespace. I think that we can merge it. Also, important thing: should we go with `ci` (or `chore`) prefix for PR's title? It's not related with source code but with infra and `infra` prefix isn't allowed and also I don't want release new version of package, or we should, WDYT?

I think `ci` works in this case.

**Comment by magicmatatjahu** [comment]
@smoya Hello! I added in the last commit the `Ingress` resource :)

**Review by smoya — APPROVED** [review]
LGTM! btw, Let's Encrypt certificate is being issued now 🚀

**Pull request merged**